packages/lodestar/src/tasks/tasks/watchEth1ForProposingTask.ts

```diff
@@ -0,0 +1,166 @@
+import {IBeaconConfig} from "@chainsafe/lodestar-config";
+import {ILogger} from "@chainsafe/lodestar-utils";
+import {IBeaconDb} from "../../db";
+import {IEth1Notifier, IDepositEvent} from "../../eth1";
+import {toHexString, fromHexString} from "@chainsafe/ssz";
+import pipe from "it-pipe";
+import abortable from "abortable-iterator";
+import {AbortController} from "abort-controller";
+import {IBeaconSync} from "../../sync";
+import {Eth1Data} from "@chainsafe/lodestar-types";
+
+export interface IWatchEth1ForProposingModules {
+  logger: ILogger;
+  db: IBeaconDb;
+  eth1: IEth1Notifier;
+  sync: IBeaconSync;
+}
+export class WatchEth1ForProposingTask {
```

Comment:
Don't really like this as a task. It should go either in eth1 module or chain or separate module. Not really sure